### PR TITLE
Filtering on activation state of workflows

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
# TL;DR

Earlier behavior of `flytectl get workflows -p project -d domain` would call ListWorkflows api which would fetch all the workflows along with all versions which was incorrect

Instead now we use ListNamedEntities api to get unique workflows  without the version 
And allow the user to fetch all versions , latest version, or by given version of workflow  by specifying the name of the workflow 

### Active state
```
flytectl get workflow -p flytesnacks -d development --filter.fieldSelector "state=0"  --filter.sortBy "created_at" 
 --------------- ------------- ------------------------------------------------------------- ------------- ------- 
| PROJECT (45)  | DOMAIN      | NAME                                                        | DESCRIPTION | STATE |
 --------------- ------------- ------------------------------------------------------------- ------------- ------- 
| flytesnacks   | development | core.containerization.use_secrets.my_secret_workflow        |             |       |
 --------------- ------------- ------------------------------------------------------------- ------------- ------- 
| flytesnacks   | development | core.control_flow.chain_tasks.chain_tasks_wf                |             |       |
 --------------- ------------- ------------------------------------------------------------- ------------- ------- 
```

### By name 
```
flytectl get workflow -p flytesnacks -d development --filter.fieldSelector "name=my.imperative.workflow.example"
 ------------- ------------- -------------------------------- ------------- ------- 
| PROJECT     | DOMAIN      | NAME                           | DESCRIPTION | STATE |
 ------------- ------------- -------------------------------- ------------- ------- 
| flytesnacks | development | my.imperative.workflow.example |             |       |
 ------------- ------------- -------------------------------- ------------- ------- 
```

### Archive state
```
flytectl get workflow -p flytesnacks -d development --filter.fieldSelector "state=1" 
 ------------- ------------- ---------------------------------------- ------------------------ ------- 
| PROJECT     | DOMAIN      | NAME                                   | DESCRIPTION            | STATE |
 ------------- ------------- ---------------------------------------- ------------------------ ------- 
| flytesnacks | development | core.containerization.raw_container.wf | raw container workflow |       |
 ------------- ------------- ---------------------------------------- ------------------------ ------- 
```

## Type
- [X] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [X] Code completed
- [X] Smoke tested
- [X] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2327

## Follow-up issue
_NA_

### CODE REVIEW

**Review by katrogan — APPROVED** [review]

File: pkg/filters/util.go [hunk 1]
```diff
@@ -29,6 +30,25 @@ func BuildResourceListRequestWithName(c Filters, project, domain, name string) (
 	return request, nil
 }
 
+func BuildNamedEntityListRequest(c Filters, project, domain string, resourceType core.ResourceType) (*admin.NamedEntityListRequest, error) {
```
[the remark this answers is not in the document]
Comment (pmahindrakar-oss) [reply pkg/filters/util.go:33]: Sure . will fix this in followup
